eclair-core/src/main/scala/fr/acinq/eclair/json/JsonSerializers.scala

```diff
@@ -402,6 +405,33 @@ object GlobalBalanceSerializer extends MinimalSerializer({
     JObject(JField("total", JDecimal(o.total.toDouble))) merge Extraction.decompose(o)(formats)
 })
 
+object OnionMessageReceivedSerializer extends MinimalSerializer({
```

Comment:
pm47 introduced a new `ConvertClassSerializer` which should be used for new JSON serializers (we should definitely do some clean-up on the old serializers and migrate them to this new mechanism).

The basic idea is to convert an `OnionMessages.ReceiveMessage` to a new `data class OnionMessageReceivedJson()` that would only contain fields which are themselves serializable.

Can you try doing this for onion messages?

Reply:
Done